src/pages/_app.tsx

```diff
@@ -35,7 +36,7 @@ export default function App({ Component, pageProps, layoutProps }) {
    * hydrate product data from page components. Ideally, we won't have to do
    * this we use more dynamic routes.
    */
-  if (process.env.HASHI_ENV !== 'production') {
+  if (process.env.HASHI_ENV !== 'production' && !Cookies.get('io_preview')) {
```

Comment:
@BRKalow does this look right?

```suggestion
  const hasPreviewCookie = typeof window !== 'undefined' && Cookies.get('io_preview')
  if (process.env.HASHI_ENV !== 'production' && !hasPreviewCookie) {
```

Reply:
Yes, but this introduces an unnecessary cookie read on each render of _app on the client, even in production. Ideally we'd move it into the `HASHI_ENV` check.